File: momhelp/src/main/java/com/momhelp/service/AiRecipeService.java
package com.momhelp.service;

import com.momhelp.dto.AiRecipeRequestDTO;
import com.momhelp.dto.AiRecipeResponseDTO;
import java.util.List;

public interface AiRecipeService {

	// Generate new recipe using AI
	AiRecipeResponseDTO generateRecipe(AiRecipeRequestDTO requestDTO);

	// Get recipe by ID
	AiRecipeResponseDTO getRecipeById(Long id);

	// Get all generated recipes
	List<AiRecipeResponseDTO> getAllRecipes();

	// Get favorite recipes
	List<AiRecipeResponseDTO> getFavoriteRecipes();

	// Mark recipe as favorite
	AiRecipeResponseDTO toggleFavorite(Long id);

	// Delete recipe
	void deleteRecipe(Long id);

	// Get recent recipes
	List<AiRecipeResponseDTO> getRecentRecipes();
}

File: momhelp-frontend/momhelp-frontend/src/services/aiRecipeService.js
import api from './api';

const aiRecipeService = {
  // Generate new recipe using AI
  generateRecipe: (recipeData) => {
    return api.post('/ai-recipe/generate', recipeData);
  },

  // Get recipe by ID
  getRecipeById: (id) => {
    return api.get(`/ai-recipe/${id}`);
  },

  // Get all generated recipes
  getAllRecipes: () => {
    return api.get('/ai-recipe/all');
  },

  // Get favorite recipes
  getFavoriteRecipes: () => {
    return api.get('/ai-recipe/favorites');
  },

  // Toggle favorite status
  toggleFavorite: (id) => {
    return api.put(`/ai-recipe/${id}/favorite`);
  },

  // Delete recipe
  deleteRecipe: (id) => {
    return api.delete(`/ai-recipe/${id}`);
  },

  // Get recent recipes
  getRecentRecipes: () => {
    return api.get('/ai-recipe/recent');
  }
};

export default aiRecipeService;